src/main/java/org/opentripplanner/routing/impl/DefaultFareServiceImpl.java

```diff
@@ -156,7 +163,29 @@ public Fare getCost(GraphPath path) {
                 currency = Currency.getInstance(fareRules.iterator().next().getFareAttribute().getCurrencyType());
             }
             hasFare = populateFare(fare, currency, fareType, rides, fareRules);
+
+            // Add the ride's fare component objects to the itinerary's fare details
+            int rideIndex = 0;
+            int legIndex = 0;
+            List<FareComponent> fareComponents = new ArrayList<FareComponent>(); // List of components for this fareType
+            if(legs != null) {
+                for (Leg leg : legs) {
+                    if (leg.isTransitLeg()) { // Match transit legs to the rides this way. Is there a better way?
```

Comment:
After looking at how this works in OTP2, we are no longer too concerned with how stable this is since the way it can be done in OTP2 is waaay better. Not going to put too much effort into this since it's pretty temporary.